src/contents/private-smart-contracts-with-solidity-and-circom.mdx

```diff
@@ -0,0 +1,370 @@
+---
+name: "Private Smart Contracts with Solidity & Circom"
+index: 1
+summary: Get started with developing privacy applications by combining Circuits and Smart Contracts.
+author: FilosofiaCodigo
+authorIcon: https://avatars.githubusercontent.com/u/707484?s=96&v=4
+authorLink: https://x.com/FilosofiaCodigo
+published: Sep 04, 2024
+readTime: 15 min read
+labels: ["ZK"]
+---
+
+ZK enables the development of applications with both private data and private execution. This opens the door to many new use cases, like the one we'll create in this guide: an anonymous and secure voting system combining Circom and Solidity.
+
+## Circom and dependencies
+
+If you don't have Circom installed yet, install it with the following commands. I'm using node v20, but it should work with other versions as well.
+
+```bash
+curl --proto '=https' --tlsv1.2 https://sh.rustup.rs -sSf | sh
+git clone https://github.com/iden3/circom.git
+cd circom
+cargo build --release
+cargo install --path circom
+npm install -g snarkjs
+```
+
+We'll also use the Circom libraries where the Poseidon function that we'll be using is located.
+
+```bash
+git clone https://github.com/iden3/circomlib.git
+```
+
+## 1. Public key creation
+
+The method we will use to conduct anonymous and secure voting is by proving that we are part of a group without revealing our identity. For example, I will vote for the president of Honduras, demonstrating that I am a Honduran without revealing which specific Honduran I am. This is called "proof of inclusion in a set."
+
+The most practical way to achieve this in zk and blockchain is through Merkle trees. We will place the voters as leaves in the tree and prove that we are one of them without disclosing which one.
+
+Since the tree is public, we will use a set of public-private key pairs so that each voter can cast their vote only once.
+
+You might wonder if we can use the public keys from our Ethereum wallet (e.g., from MetaMask). In future guides like this one, I'll address that topic just as I did with noir. To reach that point, you'll need the fundamentals from this guide. So stay tuned and subscribe!
```

Comment:
There's no subscribe, so maybe point them to follow our X account x.com/levelupweb3